### PR TITLE
Triple click template problem

### PR DESCRIPTION
**Pull request by Stelminator**
First draft of trying to fix https://github.com/lay295/TwitchDownloader/issues/762 / https://github.com/lay295/TwitchDownloader/pull/783 / https://github.com/lay295/TwitchDownloader/pull/765 / https://github.com/lay295/TwitchDownloader/issues/781

The problem seems to be with the template, not the rest of the style. Not sure if I missed some nuance there.

I just wanted to call attention to what seems to be a strict improvement; I'll try to do another pass later, but this is where I stop for now.

### CODE REVIEW

**Comment by Stelminator** [comment]
I added some WIP commits to actively use the grid layout. I'm still unfamiliar with all the pieces of WPF that are involved here, but, to me, this looks AND feels better.

* added + assigned components grid column and rows
* adjusted margins
* aligned text content to Top
* Auto vertical scroll bar
* removed textbox `HorizontalAlignment="Left"`, the lack of which seems to mean "auto expand" in this context
* removed textbox Height. It should fill the empty space, so, why specify?

image

**Comment by Stelminator** [comment]
marked ready for review, but would like some discussion and opportunity to clean it up more if any of this might get merged.

**Comment by ScrubN** [comment]
I am thoroughly confused. In my testing of #765, I found that the triple click behavior would override the default template requiring me to manually specify the template to use, but now it is no longer required?? Perhaps partial rebuild was to blame?

File: TwitchDownloaderWPF/WindowUrlList.xaml [hunk 1]
```diff
@@ -4,16 +4,30 @@
         xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
```
Comment (ScrubN) [inline TwitchDownloaderWPF/WindowUrlList.xaml:4]: Could you convert this file from tabs to 4 spaces? Most of the files are 4 spaces but it seems some are still tabs.

Comment (Stelminator) [reply TwitchDownloaderWPF/WindowUrlList.xaml:4]: That's my preference as well, just didn't want to change the status quo; will remove the conversion back to tabs.

File: TwitchDownloaderWPF/WindowUrlList.xaml [hunk 2]
```diff
@@ -4,16 +4,30 @@
...
+			<ColumnDefinition Width="*" />
+		</Grid.ColumnDefinitions>
+		<Grid.RowDefinitions>
+			<RowDefinition Height="Auto" />
+			<RowDefinition Height="1*"/>
+			<RowDefinition Height="Auto" />
+		</Grid.RowDefinitions>
+		<Button Grid.Row="2" x:Name="btnQueue" Content="{lex:Loc AddToQueue}" HorizontalAlignment="Center" VerticalAlignment="Bottom" MinWidth="110" Height="45" Margin="10,10,10,10" Click="btnQueue_Click" Background="{DynamicResource ActionButtonBackground}" Foreground="{DynamicResource ActionButtonText}" BorderBrush="{DynamicResource ActionButtonBorder}"/>
+		<Label Grid.Row="0" Content="{lex:Loc ListOfVodsClips}" HorizontalAlignment="Left" Margin="10,10,10,0" VerticalAlignment="Top" Background="{DynamicResource AppElementBackground}" BorderBrush="{DynamicResource AppElementBorder}" Foreground="{DynamicResource AppText}"/>
+		<TextBox Grid.Row="1" x:Name="textList" AcceptsReturn="True" Margin="10,10,10,0" VerticalContentAlignment="Top" VerticalScrollBarVisibility="Auto" TextWrapping="Wrap" Text="" Background="{DynamicResource AppElementBackground}" BorderBrush="{DynamicResource AppElementBorder}" Foreground="{DynamicResource AppText}"/>
```
[the remark this answers is not in the document]
Comment (Stelminator) [reply TwitchDownloaderWPF/WindowUrlList.xaml:31]: I'll play around with it. Is there a way to convert that to a variable to DRY it up?

File: TwitchDownloaderWPF/WindowUrlList.xaml [hunk 3]
```diff
@@ -4,16 +4,30 @@
...
-    </Grid>
+		<Grid.ColumnDefinitions>
+			<ColumnDefinition Width="*" />
+		</Grid.ColumnDefinitions>
+		<Grid.RowDefinitions>
+			<RowDefinition Height="Auto" />
+			<RowDefinition Height="1*"/>
+			<RowDefinition Height="Auto" />
+		</Grid.RowDefinitions>
+		<Button Grid.Row="2" x:Name="btnQueue" Content="{lex:Loc AddToQueue}" HorizontalAlignment="Center" VerticalAlignment="Bottom" MinWidth="110" Height="45" Margin="10,10,10,10" Click="btnQueue_Click" Background="{DynamicResource ActionButtonBackground}" Foreground="{DynamicResource ActionButtonText}" BorderBrush="{DynamicResource ActionButtonBorder}"/>
```
[the remark this answers is not in the document]
Comment (Stelminator) [reply TwitchDownloaderWPF/WindowUrlList.xaml:29]: Another thing I agree with and had in a draft but didn't want to go moving things for no reason. Theorized that they were in order of importance or creation, but, 🤷‍♂️, order of appearance makes sense to me. Will adjust.

File: TwitchDownloaderWPF/WindowUrlList.xaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ScrubN) [reply TwitchDownloaderWPF/WindowUrlList.xaml]: I wouldn't bother with putting margins in a variable, especially when it's only 3 elements being styled.

[the remark this answers is not in the document]
Comment (Stelminator) [reply TwitchDownloaderWPF/WindowUrlList.xaml]: I went with 6, and it looked slightly more like the elements went together than with 7.

**Comment by Stelminator** [comment]
I think I got everything mentioned.

**Review by ScrubN — APPROVED** [review]
LGTM

**Comment by ScrubN** [comment]
Thanks for double checking the templates. I still don't know why it was behaving weirdly for me the first time. Again, my only guess would have to be partial rebuild weirdness.

**Pull request merged**